install-latest.sh

```diff
@@ -389,7 +397,37 @@ get_binary_name() {
   esac
   echo "$name"
 }
+# Check if one version is less than another.
+# Versions are expected to be of the for X.X.X where X is some integer.
+# 
+# Given version strings X1.X2.X3 and Y1.Y2.Y3:
+# A version is less than another if there exists an Xn/Yn pair where Xn < Yn where for every preceding Xn/Yn pair in the sequence Xn == Yn.
+# Example:
+# 2.1.4 < 3.1.4
+# 3.5.0 < 3.6.5
+# 3.5.5 < 3.5.15
+#
+# Returns 0 when the first version argument < the second.
+# Returns 1 otherwise.
+version_less_than() {
+    for i in 1 2 3
+    do
+        VERSION_PIECE1=$(echo "$1" | cut -d '.' -f $i)
```

Comment:
First, I think I was wrong. I wrote a test harness and it worked properly with `return 1` at the end. I think I was thinking "1 is true", but this is bash so `0` is true. Is that right?

But, I think we do have a problem because we are comparing strings so we'll get things like "3.10.1" < "3.4.7".

```bash
#!/bin/sh
set -e
version_less_than() {
    for i in 1 2 3
    do
        VERSION_PIECE1=$(echo "$1" | cut -d '.' -f $i)
        VERSION_PIECE2=$(echo "$2" | cut -d '.' -f $i)

        if [ "$VERSION_PIECE1" -lt "$VERSION_PIECE2" ]
        then
            return 0
        fi

        if [ "$VERSION_PIECE1" -gt "$VERSION_PIECE2" ]
        then
            return 1
        fi
    done

    return 1
}

test() {
  if version_less_than $1 $2
  then
    echo "$1 < $2"
  else
    echo "$1 NOT < $2"
  fi
}


test "3.4.6" "3.4.7"
test "3.4.7" "3.4.7"
test "2.11.1" "3.4.7"
test "3.11.1" "3.4.7"
```

```
3.4.6 < 3.4.7
3.4.7 NOT < 3.4.7
2.11.1 < 3.4.7
3.11.1 NOT < 3.4.7
```

Reply:
1. You're right, when writing this I initially made that same mistake. Thinking that 1 was true and 0 was false, but 0/1 indicate the presence or absence of an error in sh.
2. I initially thought that `-lt` was numeric, but looking at the manpage I see that's I messed that up. Let me see what the better way to do that is. I got errors with `<` not reliably being part of `sh`, so let me see if there's some other way.

Reply:
I'm wrong here, again. Chris had it correct from the beginning, as the test case clearlly shows

Reply:
After discussing this with Scott, it looks like everything is ok.